### ISSUE TITLE
Getting Uncaught Error when using Keyboard Navigation around a disabled soccer ball

### ISSUE THREAD

**Issue by Ashton-Morris** [phetsims/mean-share-and-balance#351]
**Test Device**
MacBook Pro (16-inch, 2019)

**OS**
Monterey 12.7.6

**Browser** 
 Chrome 127.0.6533.120 
 
 **Problem Description**
Getting an error "_Error from sim: Uncaught Error: Assertion failed: reentry detected, value=[object Object], oldValue=[object Object]_" when using keyboard navigation in PhET-iO-Studio on the Balance screen. 

**Steps to Reproduce**

1. Navigate to In studio on Balance Point Screen
2. Kick the maximum number of balls
3. Option/Alt click a ball on top (this worked with balls 2 high)
4. Under that ball deselect the enabledProperty
5. Use keyboard navigation to move another ball on top of or over the ball that is not enabled
6. Get a pop up with this error "Error from sim: Uncaught Error: Assertion failed: should not change both the selection and the value when sorting"
7. If you hit ok and try to use the arrow keys with keyboard navigation again, the same error tends to pop up elsewhere.

**Visuals**

https://github.com/user-attachments/assets/59580630-7ce5-43e7-9559-8a738aef5302

![Screen Shot 2024-08-21 at 2 38 10 PM](https://github.com/user-attachments/assets/77af0478-f679-4db0-87a7-d20f1012ceda)
![Screen Shot 2024-08-21 at 2 43 19 PM](https://github.com/user-attachments/assets/9355a0a6-19b1-49f5-8924-d892832cad80)

**Code from my Console**
```
mean-share-and-balan…yEvents=false:15996 Uncaught 
Error: Assertion failed: should not change both the selection and the value when sorting
    at window.assertions.assertFunction (mean-share-and-balan…nts=false:15996:780)
    at mean-share-and-balan…false:16008:2213478
    at a.notifyLoop (mean-share-and-balan…s=false:16008:31333)
    at a.emit (mean-share-and-balan…s=false:16008:30786)
    at r._notifyListeners (mean-share-and-balan…s=false:16008:23075)
    at r.unguardedSet (mean-share-and-balan…s=false:16008:22440)
    at r.set (mean-share-and-balan…s=false:16008:22244)
    at set value (mean-share-and-balan…s=false:16008:18802)
    at mean-share-and-balan…false:16008:4516154
    at p.notifyLoop (mean-share-and-balan…s=false:16008:31333)
    at p.emit (mean-share-and-balan…s=false:16008:31188)
    at y.emit (mean-share-and-balan…s=false:16008:10119)
    at Sx.reorganizeStack (mean-share-and-balan…false:16008:4458592)
    at mean-share-and-balan…false:16008:4454447
    at a.notifyLoop (mean-share-and-balan…s=false:16008:31333)
    at a.emit (mean-share-and-balan…s=false:16008:30786)
    at nx._notifyListeners (mean-share-and-balan…s=false:16008:23075)
    at nx.unguardedSet (mean-share-and-balan…s=false:16008:22440)
    at nx.set (mean-share-and-balan…s=false:16008:22244)
    at nx.set (mean-share-and-balan…s=false:16008:18859)
```

**Comment by Nancy-Salpepi** [comment]
tagging https://github.com/phetsims/qa/issues/1137

**Comment by Nancy-Salpepi** [comment]
I was also able to reproduce with macOS 14.6.1.

**Comment by Nancy-Salpepi** [comment]
This error happens in rc.3. but not in rc.2, although in rc.2 the focus randomly changes between balls when following the above steps.

**Comment by KatieWoe** [comment]
I'm not getting this in Win 11 Chrome for some reason.

**Comment by KatieWoe** [comment]
With help from @jbphet I was able to reproduce this on Win 11 Chrome. It seems to hinge on two things from what I can tell. First, the ball on top of the 2 stack has to have been kicked after the one on the bottom. Second, the ball moving over the tower and causing the error must have been kicked before the top ball and be coming from the left. Using the dev query parameter can help with the set up. These conditions may not be exact, but hopefully can narrow things down.

**Comment by jbphet** [comment]
I tracked down the area of code where the problem was occurring and than enlisted @zepumph to help me figure out what was missing from the RC.  We (mostly he) found what we think was the culprit and I've cherry picked that only the needed branch on soccer-common.  This should be ready for the next round of testing.

**Comment by KatieWoe** [comment]
I am no longer able to reproduce this using the steps from https://github.com/phetsims/mean-share-and-balance/issues/351#issuecomment-2305803193
